docs/proposals/build-skipping.adoc

```diff
@@ -0,0 +1,55 @@
+= Proposal: Multi build repo skipping
+
+== Purpose
+
+Skip builds for which nothing has changed in their working directory.
+
+== Background
+
+For a repo with a single build, the pipeline build will only be triggered if any changes trigger a new pipeline run. 
+
+However if one has multiple builds in one repo, the pipeline build runs all build tasks even for those areas where nothing has changed. 
+
+A simple scenario is a repo with a backend and frontend component whose build outputs are combined by a subsequent docker build in a subsequent deploy task. 
+
+There is no good reason when only the frontend code changes that the backend build is redone as well and vice versa of course.  
+
+== Solution
+
+`ods-start` provides a file called `$WORKING_DIR/.ods/git-dir-commit-sha` which contains the git commit sha of the working directory (no whitespace or newlines).
+
+Build tasks check first whether their artifacts are available under the working dir commit sha. 
```

Comment:
> their artifacts

with this you mean e.g. the built binary?

Reply:
Yes, although it would only need to be a single file/artifact if we used something like nexus. 
If the artifact resides in a cache it, the build output could simply be the files in a directory with the working-dir commit as a name.